invenio_communities/assets/semantic-ui/js/invenio_communities/members/components/Dropdowns.js

```diff
@@ -134,8 +134,20 @@ export class MemberDropdown extends Component {
       value: initialValue,
     };
   }
+
+  onChange = (e, data) => {
+    const { updateMember } = this.props;
+
+    try {
+      updateMember(data.value);
```

Comment:
add `withCancel` and handle error?

Reply:
the withCancel is used inside the actual `updateMember` function, it is used here so that the `setstate` function won't be called if the functions errors. Something i proposed first was to make the dropdown controlled so that the parent can handle if the value will be set or not.

so that means the currentValue would also be passed down from a prop and the parent component can choose if it changes or not depending on certain conditions met, now this component handles it own state internally.